packages/backend/src/queryBuilder.ts

```diff
@@ -160,9 +160,32 @@ export const buildQuery = ({
         })
         .join('\n');
 
-    const sqlSelect = `SELECT\n${[...dimensionSelects, ...metricSelects].join(
-        ',\n',
-    )}`;
+    const filteredMetricSelects = getFilterRulesFromGroup(
```

Comment:
I suppose this will have an impact on CSV downloads and other features that assume all properties in the results are visible.

Reply:
Fixed CSVs

![image](https://user-images.githubusercontent.com/1983672/218764907-75c4a1fe-6682-4379-b074-2a914c631fe3.png)